src/Microsoft.ML.AutoML/TrainerExtensions/MultiTrainerExtensions.cs

```diff
@@ -229,4 +229,22 @@ public PipelineNode CreatePipelineNode(IEnumerable<SweepableParam> sweepParams,
                 columnInfo.LabelColumnName, columnInfo.ExampleWeightColumnName);
         }
     }
+
+    internal class ImageClassificationExtension : ITrainerExtension
+    {
+        public IEnumerable<SweepableParam> GetHyperparamSweepRanges() => null;
```

Comment:
Should at minimum sweep over the model architectures and possibly additional impactful hyperparameters.

Possible useful sweep targets:
* Model architecture (or [domain adapted models](https://en.wikipedia.org/wiki/Domain_adaptation))
* Layer training schedule, like only last layer, last N layers, or 100 epochs of last layer and 10 epochs of all layers
* NumIterations (epochs)
* Learning rate schedule

All of the above should have an Auto option to keep the current heuristics.

Reply:
thanks, as discussed for the first release we DO NOT want ANY sort of sweeping. Image Classification changes the learning rate after every epoch to reduce loss and it auto stops when it feels model has reached optimal accuracy. We plan to use RESNET 50 as the default architecture as it works well in most scenario and is also very popular.

Reply:
AutoML is a trade-off where compute time is used as a replacement for human time. The goal is to go after a high accuracy using the time the user hands us. If you train only one model we are not doing what the user asked of us. This misses the point of AutoML.

The current goal is to integrate support for images, while fitting in with the current AutoML platform.

Reply:
Hi @justinormont I'm not adding any sweeping because this was the initial plan and I will stick to the plan. I'm not saying we should not add sweeping, just that it will not be added in the first release.